keanu-project/src/main/java/io/improbable/keanu/tensor/generic/GenericTensor.java

```diff
@@ -73,6 +75,35 @@ public T getValue(int... index) {
         return this;
     }
 
+    public boolean isNull() {
+        return this.data == null;
+    }
+
+    public GenericTensor<T> setWithMaskInPlace(DoubleTensor mask, T value) {
+        if (this.getLength() != mask.getLength()) {
+            throw new IllegalArgumentException("The lengths of the tensor and mask must match, but got tensor length: " + this.getLength() + ", mask length: " + mask.getLength());
+        }
+
+        if (data == null) {
+            data = (T[]) new Object[(int) this.getLength()];
```

Comment:
It's fine if we need this cast, but are we sure it won't cause problems? Casting arrays can fail in non-intuitive ways due to covariance: https://stackoverflow.com/questions/28570877/java-covariant-array-bad